### ISSUE TITLE
Layout of ChatView not compact on keyboard typing new message

### ISSUE THREAD

**Issue by trinhvanminh2009** [bassaer/ChatMessageView#83]
ChatView UI not auto compact on keyboard typing the new message. How can I compact it? Set ChatView height = "wrap_content" It doesn't work.
Thank you!

**Comment by bassaer** [comment]
you mean you want ChatView to scroll bottom automatically when keyboard appears?
if so, try this.

```
mChatView.setAutoScroll(true);
```

**Comment by trinhvanminh2009** [comment]
@bassaer Thanks, but it's not working with `mChatView.setAutoScroll(true);`

**Comment by bassaer** [comment]
example app works.
Is this what you want to do?

![auto_scroll](https://user-images.githubusercontent.com/14991643/33508470-dc2985b2-d73d-11e7-93d2-c1d96b19a992.gif)

**Comment by trinhvanminh2009** [comment]
Yes, Exactly `  mChatView.setRightBubbleColor(ContextCompat.getColor(this, R.color.green500));
        mChatView.setLeftBubbleColor(Color.WHITE);
        mChatView.setBackgroundColor(ContextCompat.getColor(this, R.color.blueGray500));
        mChatView.setSendButtonColor(ContextCompat.getColor(this, R.color.red500));
        mChatView.setSendIcon(R.drawable.ic_action_send);
        mChatView.setRightMessageTextColor(Color.WHITE);
        mChatView.setLeftMessageTextColor(Color.BLACK);
        mChatView.setUsernameTextColor(Color.WHITE);
        mChatView.setSendTimeTextColor(Color.WHITE);
        mChatView.setDateSeparatorColor(Color.WHITE);
        mChatView.setInputTextHint("New comment...");
        mChatView.setMessageMarginTop(5);
        mChatView.setMessageMarginBottom(5);
        mChatView.setAutoScroll(true);
`
But it's not working like that

**Comment by bassaer** [comment]
Did you get any errors?

**Comment by trinhvanminh2009** [comment]
@bassaer Not error. But the layout of keyboard stack up on text view and button input message

**Comment by bassaer** [comment]
how about this 

```
final MessageView messageView = mChatView.getMessageView();
messageView.setOnKeyboardAppearListener(new MessageView.OnKeyboardAppearListener(){
    @Override
    public void onKeyboardAppeared(boolean hasChanged) {
        //Appeared keyboard
        if (hasChanged) {
            new Handler().post(new Runnable() {
                @Override
                public void run() {
                    //Scroll to end
                    messageView.scrollToEnd();
                }
            });
        }
    }
});

```

**Comment by trinhvanminh2009** [comment]
Thank you, but It doesn't work.

**Comment by bassaer** [comment]
Is ChatView scrolled automatically when it receive a message?

**Comment by trinhvanminh2009** [comment]
@bassaer Yes,

**Comment by trinhvanminh2009** [comment]
Can you use teamview or something like that to help me. thank you

**Comment by bassaer** [comment]
ok,  please send your id and pass to this my address
app.nakayama@gmail.com

**Comment by trinhvanminh2009** [comment]
Check mail, please

**Comment by trinhvanminh2009** [comment]
@bassaer Thank you so much. If you have any way to help me please let me know in here

**Comment by bassaer** [comment]
OK!

**Comment by michaelsam94** [comment]
did you try other window soft mode i see you use stateAlwaysHidden there is adjustPan and adjustResize?

**Comment by trinhvanminh2009** [comment]
@michaelsam94 What is stateAlwaysHidden. Does it available in this library ChatView or this Messages View?

**Comment by bassaer** [comment]
@trinhvanminh2009 
you can change it in your AndroidManifest.xml

this is the AndroidManifest.xml in example
https://github.com/bassaer/ChatMessageView/blob/8f387f70efddac661adeaa3bf67d1bd4c17b70b9/example/src/main/AndroidManifest.xml#L20

**Comment by trinhvanminh2009** [comment]
Yes, It's doesn't work!!!
